crowbar_framework/app/models/ceph_service.rb

```diff
@@ -106,6 +106,27 @@ def proposal_dependencies(role)
     answer
   end
 
+  # Helper to find a suitable node for ceph-mds role.
+  # ceph-mds conflicts with ceph-osd, and should not be assigned to the cluster node
+  def select_node_for_mds_role(nodes)
+    # do not modify array given by caller
+    mds_nodes = nodes.dup
+    mds_nodes.delete_if(&:nil?)
+    mds_nodes.reject! do |n|
+      node_is_valid_for_role(n.name, "ceph_mds") || n.roles.include?("pacemaker-cluster-member")
```

Comment:
This is very confusing. I assume the correct name for `node_is_valid_for_role` should be `node_is_not_valid_for_role` ... it's defined here https://github.com/crowbar/crowbar-core/blob/master/crowbar_framework/app/models/service_object.rb#L442